designs/cloudstack-multiple-endpoints.md

```diff
@@ -107,7 +107,13 @@ CAPC currently utilizes them to distribute machines across CloudStack Zones. How
 2. Cloudstack domain
 3. Cloudstack zone
 4. Cloudstack account
-5. A unique name
+5. A reference to the customer-provider credentials
+
+To add more detail to the credential ref, we will continue to accept a base64 encoded ini file as input from the customer when running eks-a. However,
+in order to avoid taking a dependency on a meaningless section of a transient file, we will modify EKS-A to also create the secrets to be used by CAPC.
+These secrets will be generated from the ini file provided by customer and applied to the cluster. CAPC will then proceed to take ownership of them as
```

Comment:
Maybe worth to mention in which scenarios this file will be required? For example: when creating a management cluster, always. But what about upgrade? What about workload clusters?

Another interesting questions is, what happens if the user provides an ini file with a "secret" that already exists? (imagine for management cluster upgrades). Will the secret be updated or should they be immutable?

Reply:
I think we would model all these decisions from what the vsphere provider currently has defined. As I understand, the secrets can be updated on every eks-a operation (create, upgrade) so I would expect the file will be required for every operation. VSphere seems to apply the generated secrets from the input credentials on every operation and ignores old ones. I suppose this isn't the cleanest approach but old secrets could be cleaned up manually in the worst case.

Reply:
Updated doc to reflect our offline discussion - EKS-A will create secrets but not modify existing ones to prevent unintentional overwrites. If users want to modify credentials, they can do so manually, see below